framework/helpers/BaseInflector.php

```diff
@@ -599,4 +600,27 @@ public static function sentence(array $words, $twoWordsConnector = null, $lastWo
                 return implode($connector, array_slice($words, 0, -1)) . $lastWordConnector . end($words);
         }
     }
+
+    /**
+     * @return string
+     */
+    private static function encoding()
+    {
+        return isset(Yii::$app) ? Yii::$app->charset : 'UTF-8';
+    }
+
+    /**
+     * The same as built-in `ucfirst`, but unicode-safe
+     *
+     * @param $string
+     * @param $encoding
+     * @return string
+     */
+    private static function mb_ucfirst($string, $encoding)
```

Comment:
Could be renamed to `ucfirst`, made public and moved to `StringHelper`.

Reply:
But overall I'm OK leaving it private. Parameter phpdoc types should be specified.